akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/PersistentBehaviors.scala

```diff
@@ -0,0 +1,97 @@
+
+  val initialState: State
+
+  def commandHandler(): CommandHandler[Command, Event, State]
+
+  def eventHandler(): EventHandler[Event, State]
+
+  /**
+   * @return A new, mutable, builder
+   */
+  protected def commandHandlerBuilder(commandClass: Class[Command]): CommandHandlerBuilder[Command, Event, State] =
+    new CommandHandlerBuilder[Command, Event, State](commandClass)
+
+  /**
+   * @return A new, mutable, builder
+   */
+  protected final def eventHandlerBuilder(eventClass: Class[Event]): EventHandlerBuilder[Event, State] =
+    new EventHandlerBuilder[Event, State](eventClass)
+
+  /**
+   * The `callback` function is called to notify the actor that the recovery process
+   * is finished.
+   */
+  def onRecoveryCompleted(ctx: ActorContext[Command], state: State): Unit = {}
+
+  /**
+   * Initiates a snapshot if the given function returns true.
+   * When persisting multiple events at once the snapshot is triggered after all the events have
+   * been persisted.
+   *
+   * `predicate` receives the State, Event and the sequenceNr used for the Event
+   */
+  def shouldSnapshot(state: State, event: Event, sequenceNr: Long): Boolean = false
+  /**
+   * The `tagger` function should give event tags, which will be used in persistence query
+   */
+  def tagsFor(event: Event): java.util.Set[String] = Collections.emptySet()
+
+  /**
+   * INTERNAL API
+   */
+  override private[akka] def untypedProps = {
+    new scaladsl.PersistentBehavior[Command, Event, State](
+      _ ⇒ persistenceId,
+      initialState,
+      (ctx, s, e) ⇒ commandHandler.apply(ctx.asJava, s, e).asInstanceOf[EffectImpl[Event, State]],
+      (s: State, e: Event) ⇒ eventHandler().apply(s, e),
+      (ctx, s) ⇒ onRecoveryCompleted(ctx.asJava, s),
+      e ⇒ tagsFor(e).asScala.toSet,
+      shouldSnapshot
+    ).untypedProps
+  }
+}
+
+object PersistentBehaviors {
+  /**
+   * Create a `Behavior` for a persistent actor.
+   */
+  def immutable[Command, Event, State >: Null](
+    persistenceId:  String,
+    initialState:   State,
+    commandHandler: CommandHandler[Command, Event, State],
+    eventHandler:   EventHandler[Event, State]): PersistentBehavior[Command, Event, State] =
+    persistentEntity(_ ⇒ persistenceId, initialState, commandHandler, eventHandler)
+
+  /**
+   * Create a `Behavior` for a persistent actor in Cluster Sharding, when the persistenceId is not known
+   * until the actor is started and typically based on the entityId, which
+   * is the actor name.
+   *
+   * TODO This will not be needed when it can be wrapped in `Behaviors.deferred`.
+   */
+  def persistentEntity[Command, Event, State >: Null](
+    persistenceIdFromActorName: String ⇒ String,
+    initialState:               State,
+    commandHandler:             CommandHandler[Command, Event, State],
+    eventHandler:               EventHandler[Event, State]): PersistentBehavior[Command, Event, State] = ???
```

Comment:
perhaps remove instead?

Reply:
this is for usage with Cluster Sharding and there we now have `entityId => Behavior`

I can remove it from scaladsl in my sharding PR, to also verify that it works

Reply:
Both of these are gone now we've gone with the class